### PR TITLE
Issue #383: Add ensure and enable parameter for forwarder service

### LINKED ISSUE #383: Add a $service_enable param to splunk::forwarder::service
Is it possible to make the service enable parameter configurable in the class splunk::forwarder::service
We need to set the service in windows servers  to automatic (delayed) instead of true
This is possible with the service resource enabled => delayed

### PR DESCRIPTION
**Pull request by cjswart**
On windows we need to set delayed for the service

#### Pull Request (PR) description
Issue_383: Make the ensure and enable parameter of the splunk forwarder service configurable
We need to set enable to delayed on windows servers

#### This Pull Request (PR) fixes the following issues
Fixes #383

### CODE REVIEW

File: manifests/forwarder/service.pp [hunk 1]
```diff
@@ -4,10 +4,15 @@
 #   sub-classes
 #
 class splunk::forwarder::service {
+  $provider = $facts['os']['family'] ? {
```
Comment (bastelfreak) [inline manifests/forwarder/service.pp:7]: why is that required? Did you run into any issues?

Comment (cjswart) [reply manifests/forwarder/service.pp:7]: When running de rspec test locally (on an ubuntu 22.04 bundle exec rake spec) the test on windows  fails with the error Provider redhat must have features 'delayed_startable'
I have no idea why the provider redhat was used for the service resource in the rspec test for windows but this seems to fix it.

File: manifests/forwarder.pp [hunk 2]
```diff
@@ -172,6 +178,8 @@
   Stdlib::Absolutepath $forwarder_homedir    = $splunk::params::forwarder_homedir,
   Stdlib::Absolutepath $forwarder_confdir    = $splunk::params::forwarder_confdir,
   String[1] $service_name                    = $splunk::params::forwarder_service,
+  String[1] $service_ensure                  = $splunk::params::forwarder_service_ensure,
```
[the remark this answers is not in the document]
Comment (cjswart) [reply manifests/forwarder.pp:181]: yes i shall fix this

File: manifests/forwarder.pp [hunk 3]
```diff
@@ -172,6 +178,8 @@
...
+  String[1] $service_enable                  = $splunk::params::forwarder_service_enable,
```
[the remark this answers is not in the document]
Comment (cjswart) [reply manifests/forwarder.pp:182]: I think only the ensure is a boolean i shall test this

File: manifests/params.pp [hunk 4]
```diff
@@ -166,6 +166,8 @@
       $enterprise_confdir              = "${enterprise_homedir}/etc"
       $forwarder_install_options       = []
       $enterprise_install_options      = []
+      $forwarder_service_ensure        = 'running'
```
[the remark this answers is not in the document]
Comment (cjswart) [reply manifests/params.pp:169]: ok i shall remove it

File: manifests/forwarder/service.pp (hunk 1 above)
Comment (cjswart) [reply manifests/forwarder/service.pp]: I removed the code and on puppet7 with ruby 2.7 it fails
Error: splunk::forwarder supported operating systems on windows-2012-x86_64 is expected to compile into a catalogue without dependency cycles

Failure/Error: it { is_expected.to compile.with_all_deps }
  error during compilation: Parameter enable failed on Service[SplunkForwarder]: Provider redhat must have features 'delayed_startable' to set 'enable' to 'delayed' (file: /home/runner/work/puppet-splunk/puppet-splunk/spec/fixtures/modules/splunk/manifests/forwarder/service.pp, line: 7)
  
  Do you have an other suggestion to fix this or shall i put the selector back in the code

File: manifests/forwarder.pp (hunk 3 above)
[the remark this answers is not in the document]
Comment (cjswart) [reply manifests/forwarder.pp:182]: Service enable property has different values on different OS'en possible values are true, false, manual, mask or delayed
So it can not be a boolean

File: manifests/forwarder/service.pp (hunk 1 above)
Comment (cjswart) [reply manifests/forwarder/service.pp]: Fixed it by using $facts['service_provider'] as provider and add the fact to the rspec on windows

**Review by ralfbosz — APPROVED** [review]

**Review by pemulder — APPROVED** [review]

**Review by kenyon — APPROVED** [review]
